cmd/serverless-init/main.go

```diff
@@ -82,7 +82,7 @@ func setupMetricAgent(tags map[string]string) *metrics.ServerlessMetricAgent {
 	config.Datadog.Set("use_v2_api.series", false)
 	metricAgent := &metrics.ServerlessMetricAgent{}
 	// we don't want to add the container_id tag to metrics for cardinality reasons
-	delete(tags, "container_id")
+	tags = tag.WithoutContainerID(tags)
```

Comment:
Instead of mutating the map that is also accessed in a goroutine by the trace agent, make a copy of it.

Reply:
just an idea, can we just change
```go
func GetBaseTagsArrayWithMetadataTags(metadata map[string]string) []string {
	tagsMap := GetBaseTagsMapWithMetadata(metadata)
	// we don't want to add the container_id tag to metrics for cardinality reasons
    delete(tagsMap, "container_id")
	return tags.MapToArray(tagsMap)
}
```
just for 'better encapsulation' ?

Reply:
actually, i might misunderstood it.  removing 'container_id' key does't seem to a general requirement for `GetBaseTagsArrayWithMetadataTags`

Reply:
Correct, we only want to remove it for the metrics agent. We want it to remain for the trace agent.